crc/aarch64/crc16_t10dif_copy_pmull.S

```diff
@@ -209,49 +203,42 @@ v_p4_l			.req	v17
 	sub	x_counter, x_counter, #64
 	cmp	x_counter, 63
 
-	dup	d_x0_h, v_x0.d[1]
-	dup	d_x1_h, v_x1.d[1]
-	dup	d_x2_h, v_x2.d[1]
-	dup	d_x3_h, v_x3.d[1]
-
-	dup	d_x0_l, v_x0.d[0]
-	dup	d_x1_l, v_x1.d[0]
-	dup	d_x2_l, v_x2.d[0]
-	dup	d_x3_l, v_x3.d[0]
-
-	ldr	q_y0, [x_src_saved, -64]
-	ldr	q_y1, [x_src_saved, -48]
-	ldr	q_y2, [x_src_saved, -32]
-	ldr	q_y3, [x_src_saved, -16]
-
-	str	q_y0, [x_dst_saved, -64]
-	str	q_y1, [x_dst_saved, -48]
-	str	q_y2, [x_dst_saved, -32]
-	str	q_y3, [x_dst_saved, -16]
-
-	pmull	v_x0_h.1q, v_x0_h.1d, v_p4_h.1d
-	pmull	v_x0_l.1q, v_x0_l.1d, v_p4_l.1d
-	pmull	v_x1_h.1q, v_x1_h.1d, v_p4_h.1d
-	pmull	v_x1_l.1q, v_x1_l.1d, v_p4_l.1d
-	pmull	v_x2_h.1q, v_x2_h.1d, v_p4_h.1d
-	pmull	v_x2_l.1q, v_x2_l.1d, v_p4_l.1d
-	pmull	v_x3_h.1q, v_x3_h.1d, v_p4_h.1d
-	pmull	v_x3_l.1q, v_x3_l.1d, v_p4_l.1d
-
-	tbl	v_y0.16b, {v_y0.16b}, v7.16b
-	tbl	v_y1.16b, {v_y1.16b}, v7.16b
-	tbl	v_y2.16b, {v_y2.16b}, v7.16b
-	tbl	v_y3.16b, {v_y3.16b}, v7.16b
-
-	eor	v_tmp1_x0.16b, v_x0_h.16b, v_x0_l.16b
-	eor	v_tmp1_x1.16b, v_x1_h.16b, v_x1_l.16b
-	eor	v_tmp1_x2.16b, v_x2_h.16b, v_x2_l.16b
-	eor	v_tmp1_x3.16b, v_x3_h.16b, v_x3_l.16b
-
-	eor	v_x0.16b, v_tmp1_x0.16b, v_y0.16b
-	eor	v_x1.16b, v_tmp1_x1.16b, v_y1.16b
-	eor	v_x2.16b, v_tmp1_x2.16b, v_y2.16b
-	eor	v_x3.16b, v_tmp1_x3.16b, v_y3.16b
+        ldp q_y0, q_y1, [x_src_saved, -64]
+        ldp q_y2, q_y3, [x_src_saved, -32]
+
+        stp q_y0, q_y1, [x_dst_saved, -64]
+        stp q_y2, q_y3, [x_dst_saved, -32]
+
+        prfm pldl2strm,  [x_src_saved, #1024]
+        prfm pldl2strm,  [x_src_saved, #1088]
```

Comment:
Is it useful ?

How about performance number with/without it ?

Reply:
crc16_t10dif_perf:
Start timed tests
crc16_t10dif_cold: runtime =    3068488 usecs, bandwidth 26776 MB in 3.0685 sec = 8726.26 MB/s
finish 0x46b7
[root@node4 isa-l]# crc/crc16_t10dif_copy_perf
crc16_t10dif_copy_perf:
Start timed tests
crc16_t10dif_copy_cold: runtime =    3006263 usecs, bandwidth 23488 MB in 3.0063 sec = 7813.05 MB/s
finish 0x46b7
[root@node4 isa-l]# crc/crc16_t10dif_op_perf
crc16_t10dif_streaming_insert_perf:
 size blk: 512, blk_ext: 520, blk data: 512, stream: 67108864
Start timed tests
crc16_t10pi_op_copy_insert_cold: runtime =    3017353 usecs, bandwidth 10536 MB in 3.0174 sec = 3491.83 MB/s